### PR TITLE
Add Linux job incl. CT to Appveyor

### PR DESCRIPTION
**Pull request by tolbrino**

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -5,31 +5,38 @@ skip_tags: false
...
       secure: KN38AnwEZQ6wMS1ktlkybV0BEvTUgnvajLAPBJWdH8I4lttpWzTkLdU+XgLoG6X//t76IbiaIdEPFKEDzgoHyY2xGIxofmS2BDtQD+I9T3o=
+    APPVEYOR_SSH_KEY: ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDDG7+Rf4XVbw3QUZwKR8UehNXXKuxgoLnHi4/VQ3w5CrOmJ0ANCoPsm34jJlBUM06Qkxi7LGjZs3TAKmzUKMKkG8K5y7xlHMVkYxrd9HVchrBCLCMB0SdWNZ3uqkmOtwkGCHSyYkXlzHT5qFMbxoJHsLsrXrp9UVTyOVPBZI/7lnmeLq6xofZJ7PjJA//NB0WgjU1Rlri7+Xo2/ZTUdq6i6TqX8oZHWJNCn3q3tPJ/gVU7/6FgxtArduiK8rq45uROAq1Vhhu7vXMgXZaMtzXKwhVx68w9GGLkm8e5cvey1rIaJFj3Gx7NsORwzVwwtIQ3FP1KI3qonmPvBcV/0icb
     MSVC_VERSION: 14.16.27023
     WIN_SDK_VERSION: 10.0.26624.0
     WIN_MSYS2_ROOT: C:\msys64
     WIN_MSYS2_CACHE: C:\msys64\var\cache\pacman\pkg
     WIN_OTP_PATH: C:\Program Files\erl
-    BUILD_PATH: /c/projects/build
+    BUILD_PATH: /c/projects/aeternity
+    ERL_EPMD_ADDRESS: 127.0.0.1
```
Comment (lucafavatella) [inline appveyor.yml]: Why is this necessary? It would be nice to be able to pass this to epmd always, not only in test.

File: appveyor.yml [hunk 2]
```diff
@@ -56,34 +63,50 @@ matrix:
...
 on_finish:
 # Set blockRdp to true to allow RDP, uses Git tag build-agent-v6.1.0%2B1300
-  - ps: $blockRdp = $false; iex ((new-object net.webclient).DownloadString('https://raw.githubusercontent.com/appveyor/ci/ab48d6e110f2feb585b6b908a72d6e7dbfb83cb3/scripts/enable-rdp.ps1'))
+  - ps: |
+      if ($isWindows) {
+        $blockRdp = $false; iex ((new-object net.webclient).DownloadString('https://raw.githubusercontent.com/appveyor/ci/ab48d6e110f2feb585b6b908a72d6e7dbfb83cb3/scripts/enable-rdp.ps1'))
+      }
+# Set APPVEYOR_SSH_BLOCK to true to allow SSH access
+  - sh: export APPVEYOR_SSH_BLOCK=false
+  - sh: curl -sflL 'https://raw.githubusercontent.com/appveyor/ci/master/scripts/enable-ssh.sh' | bash -e -
```
Comment (lucafavatella) [inline appveyor.yml]: Is this run above already?

File: ci/appveyor/linux_prepare.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+sudo apt update
+# sudo apt-get upgrade -y
+sudo apt install -y git curl libssl1.0.0 build-essential libsodium23 libsodium-dev autoconf ncurses-dev libssl-dev
```
Comment (lucafavatella) [inline ci/appveyor/linux_prepare.sh]: Would it be possible to make this as close as possible to https://github.com/aeternity/docker-builder/blob/23756220edbca600f3f7af53b0ee7c8f7127cad1/Dockerfile#L4-L7 and link that here? This way we would avoid to think maintaining two different sets of scripts.

File: ci/appveyor/linux_prepare.sh [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+PACKAGE_NAME=esl-erlang_${OTP_VERSION}-1~ubuntu~bionic_amd64.deb
+OTP_DOWNLOAD_URL=https://packages.erlang-solutions.com/erlang/esl-erlang/FLAVOUR_1_general/${PACKAGE_NAME}
+curl -fsSL -o /tmp/${PACKAGE_NAME} "$OTP_DOWNLOAD_URL"
+sudo apt remove -y erlang-base
+sudo apt autoremove -y
+sudo apt install -y libwxbase3.0-0v5 libwxgtk3.0-0v5 libsctp1
```
Comment (lucafavatella) [inline ci/appveyor/linux_prepare.sh]: Necessary? I am mentioning for making this closer to https://github.com/aeternity/docker-builder/blob/23756220edbca600f3f7af53b0ee7c8f7127cad1/Dockerfile

File: ci/appveyor/test.bat [hunk 5]
```diff
@@ -5,17 +5,26 @@
 @rem Required vars:
```
Comment (lucafavatella) [inline ci/appveyor/test.bat]: This script is still commented out, right?

**Comment by lucafavatella** [comment]
I see CI failing, will re-check later.

File: ci/appveyor/test.bat (hunk 5 above)
Comment (tolbrino) [reply ci/appveyor/test.bat]: Correct.

**Comment by tolbrino** [comment]
> I see CI failing, will re-check later.

Fixed by now.

File: ci/appveyor/linux_prepare.sh (hunk 4 above)
Comment (tolbrino) [reply ci/appveyor/linux_prepare.sh]: Required because this setup uses the precompiled package from ESL which has these package requirements.

File: ci/appveyor/linux_prepare.sh (hunk 3 above)
Comment (tolbrino) [reply ci/appveyor/linux_prepare.sh]: Its somewhat different though since Erlang is not compiled from source in this setup.

File: appveyor.yml (hunk 2 above)
Comment (tolbrino) [reply appveyor.yml]: What exactly do you mean?

File: appveyor.yml (hunk 1 above)
Comment (tolbrino) [reply appveyor.yml]: On Windows the default behaviour of epmd is somewhat undeterministic and specific to the setup. That's why I fixed it here to locahost.

File: appveyor.yml (hunk 2 above)
Comment (lucafavatella) [reply appveyor.yml]: At line 75 I see same `   - sh: curl -sflL 'https://raw.githubusercontent.com/appveyor/ci/master/scripts/enable-ssh.sh' | bash -e - `

Comment (tolbrino) [reply appveyor.yml]: It is required because the previously set var `APPVEYOR_SSH_BLOCK` will make the script behave differently than during the init phase. Same with the rdp script.

File: ci/appveyor/linux_prepare.sh [hunk 6]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
```
Comment (dincho) [inline ci/appveyor/linux_prepare.sh]: Could you please elaborate further why you need so many hacks during the install here?

While [the script for CircleCI](https://github.com/aeternity/aeternity/blob/master/.circleci/config.yml#L58-L64) is about Ubuntu 16.04, it's straight forward.

And concretely why:
- `sudo apt remove -y erlang-base` ?
- `sudo apt --fix-broken install -y` ?
- this additional dependancy `libsctp1` ?
- the specific 0v5 in `libwxbase3.0-0v5 libwxgtk3.0-0v5` ?
- `libwxbase3.0-0v5 libwxgtk3.0-0v5` is not installed at once with the other deps above ?

File: ci/appveyor/linux_prepare.sh [hunk 7]
```diff
@@ -2,14 +2,4 @@
...
-PACKAGE_NAME=esl-erlang_${OTP_VERSION}-1~ubuntu~bionic_amd64.deb
-OTP_DOWNLOAD_URL=https://packages.erlang-solutions.com/erlang/esl-erlang/FLAVOUR_1_general/${PACKAGE_NAME}
-curl -fsSL -o /tmp/${PACKAGE_NAME} "$OTP_DOWNLOAD_URL"
-sudo apt remove -y erlang-base
-sudo apt autoremove -y
-sudo apt install -y libwxbase3.0-0v5 libwxgtk3.0-0v5 libsctp1
-sudo dpkg -i /tmp/${PACKAGE_NAME} || true
-sudo apt --fix-broken install -y || true
-sudo dpkg -i /tmp/${PACKAGE_NAME}
+sudo apt install -y git curl libssl1.0.0 build-essential libsodium23 libsodium-dev autoconf ncurses-dev libssl-dev erlang
```
Comment (lucafavatella) [inline ci/appveyor/linux_prepare.sh]: erlang: which OTP version is this? This may start failing overnight (not pinned).

Comment (tolbrino) [reply ci/appveyor/linux_prepare.sh]: [OTP 20.2](https://packages.ubuntu.com/bionic/erlang)

**Comment by tolbrino** [comment]
@dincho Simpler preparation script works.

**Comment by tolbrino** [comment]
@dincho Could you review this PR?

**Comment by tolbrino** [comment]
@ThomasArts @lucafavatella I fixed a minor path typo, need reviews again.
